### PR TITLE
modify pools to also carry ekg stats

### PR DESCRIPTION
**Pull request by evertedsphere**
required for the corresponding feature in pro monitoring (hasura/graphql-engine-mono#798)

(Corresponding pro PR: https://github.com/hasura/graphql-engine-mono/pull/885)

### CODE REVIEW

File: src/Database/PG/Query/Pool.hs [hunk 1]
```diff
@@ -49,11 +50,25 @@ import qualified Data.HashTable.IO             as HI
...
+    _pool :: RP.Pool PGConn,
+    -- | EKG stats about how we acquire, release, and manage connections
+    _stats :: PGPoolStats
+  }
+
+-- | Actual ekg gauges and other metrics are not created here, since those depend on
+-- a store and it's much simpler to perform the sampling of the distribution from within graphql-engine.
+newtype PGPoolStats = PGPoolStats
+  { -- | time taken to acquire new connections from postgres
+    _connAcquireTime :: Distribution
```
Comment (ecthiender) [inline src/Database/PG/Query/Pool.hs:67]: If I understand #798 correctly, we want the time taken to acquire connections from the connection pool. Is that right?

Although, I think its a great idea to capture the time to postgres as well. We should definitely have this. But, I think we have to add another `Distribution` to track the same thing but for connections from pool.

Possibly in this function: https://github.com/hasura/pg-client-hs/blob/swann/pool-metrics/src/Database/PG/Query/Pool.hs#L273 ?

**Review by ecthiender — APPROVED** [review]

File: src/Database/PG/Query/Pool.hs [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 module Database.PG.Query.Pool
   ( ConnParams (..)
-  , PGPool
+  , PGPool(..)
+  , PGPoolStats(..)
```
Comment (vladciobanu) [inline src/Database/PG/Query/Pool.hs:11]: I would rather keep the `PGPool` hidden and provide a function `PGPool -> PGPoolStats`

**Review by vladciobanu — CHANGES_REQUESTED** [review]
I wish I had a concrete proposal of how to make this library extensible and have `EKG` be an extension in the monorepo. However, I'm not sure how to do that without providing callback functions like `beforeConnectionAcquisition`, `afterConnectionAcquisition` as `IO ()` functions, and I like that less than the current implementation.

However, I still think we should keep this as an implementation detail, so I would not export the constructor, see comment below.

**Review by vladciobanu — APPROVED** [review]
:+1:

**Pull request merged**